bug in S1068

When applying Sorald to npefix, three fields are removed which are actually required (npefix does not compile after sorald-repair)

```
[ERROR] /home/martin/martin-no-backup/npefix/src/main/java/fr/inria/spirals/npefix/resi/context/Lapse.java:[45,21] cannot find symbol
[ERROR]   symbol: variable strategySelector
[ERROR] /home/martin/martin-no-backup/npefix/src/main/java/fr/inria/spirals/npefix/resi/context/TryContext.java:[13,21] cannot find symbol
[ERROR]   symbol: variable context
[ERROR] /home/martin/martin-no-backup/npefix/src/main/java/fr/inria/spirals/npefix/transformer/processors/RemoveNullCheckProcessor.java:[97,9] cannot find symbol
[ERROR]   symbol:   variable isProcessed
[ERROR]   location: class fr.inria.spirals.npefix.transformer.processors.RemoveNullCheckProcessor
```

--- a/sorald/src/main/java/sorald/processor/UnusedPrivateFieldProcessor.java
+++ b/sorald/src/main/java/sorald/processor/UnusedPrivateFieldProcessor.java
@@ -1,12 +1,29 @@
 package sorald.processor;
 
+import java.util.List;
 import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.code.CtFieldWrite;
+import spoon.reflect.code.CtStatement;
+import spoon.reflect.declaration.CtClass;
 import spoon.reflect.declaration.CtField;
+import spoon.reflect.visitor.filter.TypeFilter;
 
 @ProcessorAnnotation(key = "S1068", description = "Unused \"private\" fields should be removed")
 public class UnusedPrivateFieldProcessor extends SoraldAbstractProcessor<CtField<?>> {
     @Override
     protected void repairInternal(CtField<?> element) {
+        removeAllUnusedPrivateFieldWrites(element);
         element.delete();
+    }
+
+    /** Removes all writes to the given field that are not read anywhere. */
+    private static void removeAllUnusedPrivateFieldWrites(CtField<?> field) {
+        CtClass<?> parentClass = field.getParent(CtClass.class);
+        List<CtFieldWrite<?>> unusedFieldWrites =
+                parentClass.filterChildren(new TypeFilter<>(CtFieldWrite.class)).list();
+        for (CtFieldWrite<?> unusedFieldWrite : unusedFieldWrites) {
+            CtStatement statement = unusedFieldWrite.getParent(CtStatement.class);
+            statement.delete();
+        }
     }
 }

--- a/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java
+++ b/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java
@@ -1,7 +1,7 @@
 public class PrivateFieldWrittenButNotRead {
-    private int x;
+    private int x; // Noncompliant
 
-    private String foo;
+    private String foo; // Noncompliant
 
     public PrivateFieldWrittenButNotRead(int x, int y) {
         this.x = x + y;

--- a/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java
+++ b/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java
@@ -1,0 +1,13 @@
+public class PrivateFieldWrittenButNotRead {
+    private int x;
+
+    private String foo;
+
+    public PrivateFieldWrittenButNotRead(int x, int y) {
+        this.x = x + y;
+    }
+
+    public void initialize(String bar) {
+        foo = bar;
+    }
+}

--- a/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java.expected
+++ b/sorald/src/test/resources/processor_test_files/S1068_UnusedPrivateField/PrivateFieldWrittenButNotRead.java.expected
@@ -1,0 +1,6 @@
+public class PrivateFieldWrittenButNotRead {
+
+    public PrivateFieldWrittenButNotRead(int x, int y) { }
+
+    public void initialize(String bar) { }
+}

